elements/lisk-dpos/src/delegates_list.ts

```diff
@@ -36,21 +57,40 @@ const shuffleDelegateListForRound = (round, list) => {
 	return delegateList;
 };
 
-class DelegatesList {
-	constructor({ storage, activeDelegates, slots, exceptions }) {
+export class DelegatesList {
+	private readonly storage: Storage;
+	private readonly slots: Slots;
+	private readonly activeDelegates: number;
+	private readonly exceptions: {
+		readonly ignoreDelegateListCacheForRounds?: ReadonlyArray<number>;
+	};
+
+	public constructor({
+		storage,
+		activeDelegates,
+		slots,
+		exceptions,
+	}: DelegatesListConstructor) {
 		this.storage = storage;
-		this.slots = slots;
 		this.activeDelegates = activeDelegates;
+		this.slots = slots;
 		this.exceptions = exceptions;
 	}
 
 	/**
 	 * Get shuffled list of active delegate public keys (forger public keys) for a specific round.
 	 * The list of delegates used is the one computed at the end of the round `r - delegateListRoundOffset`
 	 */
-	async getForgerPublicKeysForRound(round, delegateListRoundOffset, tx) {
+	public async getForgerPublicKeysForRound(
+		round: number,
+		delegateListRoundOffset?: number,
+		tx?: StorageTransaction,
```

Comment:
if this `tx` refers to a database transaction it would be better to find a better name to avoid the conflict with the term `transaction` on our domain

Reply:
oh well maybe this was already like this so it's ok if too much effort to rename now.